src/services/dsnp.ts

```diff
@@ -0,0 +1,267 @@
+ * This function must be called with a Metamask or Torus WalletType before
+ * any other DSNP functionality is used (including subscriptions).
+ * @param walletType which wallet plugin we should use to retrieve the chain provider.
+ * @throws if walletType is unrecognized or wallet plugin cannot supply a provider.
+ */
+export const setupProvider = (walletType: WalletType): void => {
+  let eth;
+
+  if (walletType === WalletType.TORUS) {
+    eth = torusWallet.getWeb3().currentProvider;
+  } else if (walletType === WalletType.METAMASK) {
+    const global: any = window;
+    eth = global.ethereum;
+
+    if (!eth) {
+      throw new Error(
+        "Could not create provider, because ethereum has not been set"
+      );
+    }
+  } else {
+    throw new Error(
+      `Unknown walletType attempting to setup provider: ${walletType}`
+    );
+  }
+
+  const provider = new providers.Web3Provider(eth);
+  setConfig({
+    provider: provider,
+    signer: provider.getSigner(),
+    store: new Store(),
+  });
+};
+
+/**
+ * getSocialIdentities looks up all DSNP registrations for a given wallet address.
+ * @param walletAddress address of an on-chain account.
+ * @return all DSNP registration (DSNP id and handle) for which the given address is authorized to sign.
+ */
+export const getSocialIdentities =
+  core.contracts.registry.getRegistrationsByWalletAddress;
+
+/**
+ * createNewDSNPRegistration creates a new DSNP identity for a given wallet address.
+ * This function will register a new DSNP Id with the provided handle and then delegate authority
+ * to managed the Id to the owner of the provided wallet address.
+ * @param addr address of an on-chain account
+ * @param handle handle to be registered with the new Id.
+ * @returns the DSNPUserURI for the new account.
+ * @throws if registration fails for any reason, or the handle is already registered.
+ */
+export const createNewDSNPRegistration = createRegistration;
+
+/**
+ * startSubscriptions starts listenting for DNSP publication events and DSNP registration change
+ * events. The subscriptions start with all previous events which will be immediately emitted to
+ * the handlers, and then will continue listening for new events until the returned unsubscribe
+ * function is called.
+ * @param batchHandler a function called when a batch announcement is made on chain.
+ * @param registryHandler a function called when a DSNP registration is made or changed.
+ * @returns an unsubscribe function that will stop the subscriptions when called.
+ */
+export const startSubscriptions = async (
+  batchHandler: BatchAnnouncementHandler,
+  registryHandler: RegistryUpdateHandler
+): Promise<UnsubscribeFunction> => {
+  let blockNumber: number;
+  let blockIndex = 0;
+
+  // subscribe to all announcements
+  const unsubscribeToBatchPublications = await core.contracts.subscription.subscribeToBatchPublications(
+    (announcement: BatchPublicationLogData) => {
```

Comment:
does this help you to simply this? I think it was recently added.

https://github.com/LibertyDSNP/sdk-ts/blob/86fe0062c07aa54632f708ccf402223fd522a5eb/src/core/contracts/utilities.ts#L80

Reply:
Nice! yes it does. There's another PR for the upgrade to 3.0 so I'll fix it there.